assets/js/hooks/useMonitorInternetConnection.test.js

```diff
@@ -0,0 +1,177 @@
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	actHook as act,
+	createTestRegistry,
+	renderHook,
+	waitForTimeouts,
+} from '../../../tests/js/test-utils';
+import { CORE_UI } from '../googlesitekit/datastore/ui/constants';
+import { useMonitorInternetConnection } from './useMonitorInternetConnection';
+
+describe( 'useMonitorInternetConnection', () => {
+	let registry;
+	let store;
+	let originalNavigatorOnline;
+
+	const mockOnlineStatus = ( status = true ) => {
+		Object.defineProperty( navigator, 'onLine', {
+			value: status,
+			writable: true,
+		} );
+	};
+
+	const healthCheckEndpoint = new RegExp(
+		'google-site-kit/v1/core/site/data/health-checks'
+	);
+
+	const healthCheckResponse = {
+		checks: {
+			googleAPI: {
+				pass: true,
+				errorMsg: '',
+			},
+			skService: {
+				pass: true,
+				errorMsg: '',
+			},
+		},
+	};
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+
+		store = registry.stores[ CORE_UI ].store;
+	} );
+
+	afterEach( () => {
+		mockOnlineStatus();
+	} );
+
+	beforeAll( () => {
+		originalNavigatorOnline = Object.getOwnPropertyDescriptor(
+			navigator,
+			'onLine'
+		);
+	} );
+
+	afterAll( () => {
+		Object.defineProperty( navigator, 'onLine', originalNavigatorOnline );
+	} );
+
+	it( 'should set online status correctly', () => {
+		fetchMock.getOnce( healthCheckEndpoint, {
+			body: healthCheckResponse,
+		} );
+
+		renderHook( () => useMonitorInternetConnection(), { registry } );
+
+		expect( store.getState().isOnline ).toBe( true );
+	} );
+
+	it( 'should set offline status correctly', () => {
+		mockOnlineStatus( false );
+
+		renderHook( () => useMonitorInternetConnection(), { registry } );
+
+		expect( store.getState().isOnline ).toBe( false );
+	} );
+
+	it( 'should correctly change connection status when going offline then online', async () => {
+		fetchMock.get( healthCheckEndpoint, {
+			body: healthCheckResponse,
+		} );
+
+		renderHook( () => useMonitorInternetConnection(), {
+			registry,
+		} );
+
+		// Wait for fetch to complete.
+		await waitForTimeouts( 100 );
+
+		// Initial status should be online.
+		expect( store.getState().isOnline ).toBe( true );
+
+		// Simulate going offline.
+		await act( async () => {
+			mockOnlineStatus( false );
+			global.window.dispatchEvent( new Event( 'offline' ) );
+			// Wait for fetch to complete.
+			await waitForTimeouts( 100 );
+		} );
+
+		expect( store.getState().isOnline ).toBe( false );
+
+		// Simulate going back online.
+		await act( async () => {
+			mockOnlineStatus();
+			global.window.dispatchEvent( new Event( 'online' ) );
+			// Wait for fetch to complete.
+			await waitForTimeouts( 100 );
+		} );
+
+		expect( store.getState().isOnline ).toBe( true );
+		expect( fetchMock ).toHaveFetched( healthCheckEndpoint );
+	} );
+
+	it( 'should check online status in correct interval when online', () => {
+		jest.useFakeTimers();
+		jest.spyOn( global, 'setInterval' );
```

Comment:
Hi @zutigrm, checking for calls to `setInterval()` is a bit too coupled to the implementation. I would suggest taking the approach of testing the observable side effects instead. For example we could do something like this:

```js
	it( 'should check online status in correct interval when online', () => {
		jest.useFakeTimers();

		fetchMock.get( healthCheckEndpoint, {
			body: healthCheckResponse,
		} );

		renderHook( () => useMonitorInternetConnection(), { registry } );

		expect( store.getState().isOnline ).toBe( true );

		// The initial fetch will happen immediately.
		expect( fetchMock ).toHaveFetchedTimes( 1 );

		jest.advanceTimersByTime( 119999 );

		expect( fetchMock ).toHaveFetchedTimes( 1 );

		jest.advanceTimersByTime( 1 );

		// The second fetch will happen after 120000ms.
		expect( fetchMock ).toHaveFetchedTimes( 2 );
	} );
```

Please can you update this test and the one below accordingly?

Reply:
Thanks, got it.

But the second test is testing the interval when use is offline, when `isOnline` is `false` it only updates the state and bails early, there is no side effect in that case, only when `isOnline` goes back online

Reply:
I was thinking, I guess absence of request is valid to test also :D